internal/pkg/controller/controller.go

```diff
@@ -0,0 +1,332 @@
+package controller
+
+import (
+	"time"
+	"fmt"
+	"strings"
+	"bytes"
+	"sort"
+	
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	informerruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/wait"
+	errorHandler "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"github.com/sirupsen/logrus"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	updateOnChangeAnnotation = "reloader.stakater.com/update-on-change"
+	// AllNamespaces as our controller will be looking for events in all namespaces
+	AllNamespaces = "temp-reloader"
+)
+
+// Event indicate the informerEvent
+type Event struct {
+	key          string
+	eventType    string
+	namespace    string
+	resourceType string
+}
+
+// Controller for checking events
+type Controller struct {
+	client        kubernetes.Interface
+	indexer       cache.Indexer
+	queue         workqueue.RateLimitingInterface
+	informer      cache.Controller
+	resource	  string
+
+	stopCh chan struct{}
+}
+
+// NewController for initializing a Controller
+func NewController(
+	client kubernetes.Interface,
+	resource string, objType informerruntime.Object) (*Controller, error) {
+
+	c := Controller{
+		client:   client,
+		resource: resource,
+		stopCh:   make(chan struct{}),
+	}
+
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+	listWatcher := cache.NewListWatchFromClient(client.CoreV1().RESTClient(), resource, AllNamespaces, fields.Everything())
```

Comment:
`AllNamespaces` variable shouldn't be used. Before running the controller, the code should be able to detect which namespace the controller is running in. See how IMC fetches the current namespace and passes it on to the controller